CollectionView/ViewController.swift

```diff
@@ -16,7 +16,53 @@ class ViewController: UIViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        let layout = UICollectionViewFlowLayout()
+        
+        collectionView.collectionViewLayout = layout
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        collectionView.register(MyCollectionViewCell.nib(), forCellWithReuseIdentifier: MyCollectionViewCell.identifier)
+        	
+    }
+}
+
+extension ViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let storyBoard: UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
+        let detailsViewController = storyBoard.instantiateViewController(withIdentifier: "DetailsViewController") as! DetailsViewController
+        
 
+        print("You tapped an item")
+        detailsViewController.imageToLoad = imageNames[indexPath.item]
+       self.navigationController?.pushViewController(detailsViewController, animated: true)
+    }
+}
+extension ViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return imageNames.count
+    }
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        imageView = UIImage(named: imageNames[indexPath.item])
+        
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: MyCollectionViewCell.identifier, for: indexPath) as? MyCollectionViewCell, let imageView = imageView else { return UICollectionViewCell() }
+        
+        cell.configure(with: imageView)
```

Comment:
As per my comment at the top of the class, since the `UIImage` object is used only in the collection view cell, it would be better if we created it there.

Reply:
Changed! 
Thank you!

Reply:
My point was actually sending the name of the image to `configure(with:)` instead of the UIImage object. I find weird doing this way but it's more of a personal thing.